python/python/lance/util.py

```diff
@@ -40,3 +45,71 @@ def sanitize_ts(ts: ts_types) -> datetime:
     elif not isinstance(ts, datetime):
         raise TypeError(f"Unrecognized version timestamp {ts} of type {type(ts)}")
     return ts
+
+
+class KMeans:
+    def __init__(self, k: int, metric_type: str = "l2", max_iters: int = 50):
+        """Create a KMeans model.
+
+        Parameters
+        ----------
+        k: int
+            The number of clusters to create.
+        metric_type: str, default="l2"
+            The metric to use for calculating distances between vectors.
+            Supported distance metrics: "l2", "cosine", "dot"
+        max_iters: int
+            The maximum number of iterations to run the KMeans algorithm. Default: 50.
+        """
+        self.k = k
+        self._metric_type = metric_type
+        self._kmeans = _KMeans(k, metric_type, max_iters=max_iters)
+
+    def __repr__(self) -> str:
+        return f"lance.KMeans(k={self.k}, metric_type={self._metric_type})"
+
+    @property
+    def centroids(self) -> Optional[pa.FixedSizeListArray]:
```

Comment:
Should we return this as a fixed-shape tensor array instead? Then the user will have a method to convert to numpy if they want: https://arrow.apache.org/docs/dev/python/generated/pyarrow.FixedShapeTensorArray.html#pyarrow.FixedShapeTensorArray.to_numpy_ndarray

Reply:
Done

Reply:
Yeah that's fine for now then.